app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -235,19 +234,46 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
 
     override fun dispatchKeyEvent(event: KeyEvent): Boolean {
         val fragmentManager = supportFragmentManager
-        val maybeBrowserFragment = (fragmentManager.findFragmentByTag(WebRenderFragment.FRAGMENT_TAG) as WebRenderFragment?)?.let {
+
+        val maybeBrowserFragment = (fragmentManager.findFragmentByTag(WebRenderFragment.FRAGMENT_TAG)
+                as WebRenderFragment?)?.let {
             if (it.isVisible) it else null
         }
-        val maybePocketFragment = (fragmentManager.findFragmentByTag(PocketVideoFragment.FRAGMENT_TAG) as PocketVideoFragment?)?.let {
-            if (it.isVisible) it else null
+
+        if (event.keyCode == KeyEvent.KEYCODE_MENU) {
+            if (event.action == KeyEvent.ACTION_DOWN) {
+                showOrHideOverlay()
+            }
+            return true
         }
+
         return videoVoiceCommandMediaSession.dispatchKeyEvent(event) ||
-                (maybePocketFragment?.dispatchKeyEvent(event) ?: false) ||
                 (maybeBrowserFragment?.dispatchKeyEvent(event) ?: false) ||
                 super.dispatchKeyEvent(event)
     }
 
-    companion object {
-        const val PARENT_FRAGMENT = "PARENT_FRAGMENT"
+    /**
+     * If overlay is shown, webRenderFragment will be shown
+     */
+    fun showOrHideOverlay() {
+        val maybeBrowserFragment = (supportFragmentManager.findFragmentByTag(WebRenderFragment.FRAGMENT_TAG)
```

Comment:
This is verbose and repeated a few times. You could trim it down with an ext fn like: `FragmentManager.getBrowserFragment()`: https://github.com/mozilla-mobile/firefox-echo-show/blob/986826e0782b3df4babde905b4549feec438d28f/app/src/main/java/org/mozilla/focus/ext/FragmentManager.kt#L10

And maybe another: `Fragment.getVisibleOrNull()`